partner_phone_validation/res_partner_with_phone_validation.py

```diff
@@ -0,0 +1,22 @@
+from odoo import api, models
+
+
+class Contact(models.Model):
+    """This class was copied from the module crm_phone_validation.
+
+    The original code can be found here:
+    https://github.com/odoo/odoo/blob/12.0/addons/crm_phone_validation/models/res_partner.py
+    """
+
+    _name = 'res.partner'
+    _inherit = ['res.partner', 'phone.validation.mixin']
+
+    @api.onchange('phone', 'country_id', 'company_id')
+    def _onchange_phone_validation(self):
+        if self.phone:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Access to member 'phone' before its definition line 17](https://app.codacy.com/app/numigi/odoo-partner-addons/pullRequest?prid=2454434)